src/components/Table.js

```diff
@@ -274,21 +446,41 @@ export default class Table extends React.Component<TableDefaultProps, TableProps
   }
 
   render() {
+    const { multiSelect, selection, activeRowId, anchorRowBackgroundColor } = this.props;
+    const { selectedRowIndices } = this.state;
+
     let selectedRowIds;
-    if (Array.isArray(this.props.selection)) {
-      selectedRowIds = this.props.selection;
-    } else if (typeof this.props.selection === 'string') {
-      selectedRowIds = [this.props.selection];
+    if (Array.isArray(selection)) {
+      selectedRowIds = selection;
+    } else if (typeof selection === 'string') {
+      selectedRowIds = [selection];
     } else {
       selectedRowIds = [];
     }
 
+    // FIXME: Move data manipulation outside of render function.
+    // If multiSelect include row index selections made using keyboard
```

Comment:
Shouldn't all the rows that are selected be in `this.props.selection`? The source of truth for the selection is the property that the parent passes in, right? I thought that the only thing our state keeps is information about the anchor row and the active row, but that those rows should be in the selection array.

Reply:
The <Table /> component should be usable as is without requiring a great deal of effort from whatever component renders it. `onSelect` is a convenience function. Using indices for selection tracking is far more performant because it's quicker to derive any value if you have the index, but if you just have the id, you have to search for the row, then derive what you want, instead of just grabbing it. However, now I know I can ignore backwards compatibility for MasterDetails and Table, it makes sense to keep these aligned and pass index through onSelect instead of id. It's trivial to derive id or row or whatever from the index. That will save the need to do the search in the render function of the MasterDetails component. It's unfortunate we're stuck with the inner-workings of react bootstrap table for now, it still requires an array of ids. I've worked on far more complex selection scenarios with larger data sets where performance was more important, and using index for selection in that scenario becomes a necessity. We don't know how this component will be used in the future, so we should follow best practices instead of using id which requires a search and find for every selection. When you have control of your own table, individual rows can determine if they are active or not, eliminating the need for some of these gymnastics that react-bootstrap requires, but for now, it's handy that it provides sorting and filtering out of the box for us in a way that appears to be meeting our needs for the moment.

It would also make these components more composable if the logic in `componentWillReceiveProps` in `MasterDetails` were moved into `<Table />` where `onSelect` would be called, if specified, so I'll make that change as well to make Table easier to use as a standalone.

Reply:
Also worth noting that our current version of react-bootstrap-table is deprecated. react-bootstrap-table2 exposes `rowIndex` in the `onSelect` function.